pkgs/tools/graphics/argyllcms/default.nix

```diff
@@ -25,109 +101,60 @@ stdenv.mkDerivation rec {
 
-      # enable CMF Measurement device and accessory support (if present)
-      USE_CMFM = false ;
+      export AR="$AR rusc"
 
-      # Use ArgyllCMS version of libusb (deprecated - don't use)
-      USE_LIBUSB = false ;
-
-      # Compile in graph plotting code (Not fully implemented)
-      USE_PLOT = true ;		# [true]
-
-      JPEGLIB = ;
-      JPEGINC = ;
-      HAVE_JPEG = true ;
-
-      TIFFLIB = ;
-      TIFFINC = ;
-      HAVE_TIFF = true ;
-
-      PNGLIB = ;
-      PNGINC = ;
-      HAVE_PNG = true ;
-
-      ZLIB = ;
-      ZINC = ;
-      HAVE_Z = true ;
-
-      SSLLIB = ;
-      SSLINC = ;
-      HAVE_SSL = true ;
-
-      LINKFLAGS +=
-        ${lib.concatStringsSep " " (map (x: "-L${x}/lib") buildInputs)}
-        -lrt -lX11 -lXext -lXxf86vm -lXinerama -lXrandr -lXau -lXdmcp -lXss
-        -ljpeg -ltiff -lpng -lssl ;
-    '';
-  in ''
-    cp ${jamTop} Jamtop
-    substituteInPlace Makefile --replace "-j 3" "-j $NIX_BUILD_CORES"
-    # Remove tiff, jpg and png to be sure the nixpkgs-provided ones are used
-    rm -rf tiff jpg png
-
-    export AR="$AR rusc"
+      cat "$extraJamtopPath" >> Jamtop
+  '' + lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+    substituteInPlace Jambase --replace-fail "-m64" ""
   '';
 
-  buildInputs = [
-    libtiff libjpeg libpng libX11 libXxf86vm libXrandr libXinerama libXext
-    libXrender libXScrnSaver libXdmcp libXau openssl
-  ];
-
-  buildFlags = [ "all" ];
+  extraJamtop =
+    let
+      commonLinkFlags = "-ljpeg -lpng -ltiff -lusb-1.0 -lz -lssl";
+      markPresent = name: ''
+        HAVE_${name} = true ;
+        ${name}INC = ;
+        ${name}LIB = ;
+      '';
+    in
+    ''
+      ${markPresent "JPEG"}
+      ${markPresent "PNG"}
+      ${markPresent "SSL"}
+      ${markPresent "TIFF"}
+      ${markPresent "Z"}
+      ${markPresent "ZLIB"}
+
+      ${lib.optionalString stdenv.isLinux ''
+        LINKFLAGS = ${lib.concatMapStringsSep " " (dep: "-L${lib.getOutput "lib" dep}/lib") buildInputs} ;
+        LINKFLAGS += -ldl -lrt ${commonLinkFlags} ;
+        GUILINKFLAGS = -lX11 -lXext -lXxf86vm -lXinerama -lXrandr -lXau -lXdmcp -lXss ;
+      ''}
+
+      ${lib.optionalString stdenv.isDarwin ''
+        LINKFLAGS += ${commonLinkFlags} ;
+      ''}
+    '';
 
-  makeFlags = [
-    "PREFIX=${placeholder "out"}"
-  ];
+  passAsFile = [ "extraJamtop" ];
 
-  # Install udev rules, but remove lines that set up the udev-acl
-  # stuff, since that is handled by udev's own rules (70-udev-acl.rules)
   postInstall = ''
-    rm -v $out/bin/License.txt
-    mkdir -p $out/etc/udev/rules.d
-    sed -i '/udev-acl/d' usb/55-Argyll.rules
-    cp -v usb/55-Argyll.rules $out/etc/udev/rules.d/
-
     sed -i -e 's/^CREATED .*/CREATED "'"$(date -d @$SOURCE_DATE_EPOCH)"'"/g' $out/share/argyllcms/RefMediumGamut.gam
 
+    ${lib.optionalString stdenv.isLinux ''
+      install -Dm444 udev-rules/55-Argyll.rules $out/etc/udev/rules.d/55-Argyll.rules
+    ''}
   '';
```

Comment:
A few things:
1. it's now the 3rd round of review. I understand reviewers can miss details on the first round (I am guilty of this too), however both of our time is precious and I would wish to receive all feedback related to a change in the least number of batches possible (down to 1 if at all possible).
2. 80% of the feedback I received it's purely stylistic and not functionally changing the derivation in any way. `install -D` vs `mkdir + cp`, string concatenation instead of interpolation (and extra parentheses). A quick search reveals that `nixpkgs` contains all the patterns I used in other derivations and I don't see anywhere outlined which style is recommended / preferred / enforced.
3. Specifically for these cases of string interpolation, I prefer the derivations to have a easy way to change the order in which things happen, with the least amount of keystrokes possibles if you wish, so giving up on interpolation and going back to concatenation always means having to move around the plusses and the quotes. Especially on single line additions, particular care must be taken to ensure the final newline char is there, or use a multiline string delimited by `''`. It eventually turns out less readable for me.

Reply:
Thanks for the feedback, @zaninime, all valid points.

Re 1.:
At times I find working in phases helpful for larger PRs.

Re 2.:
The given suggestions are just that: suggestions. From my point of view PR authors can disagree with the suggestions, provide arguments against them to convince reviewers why things should be as proposed, and choose to accept or disregard suggestions 🙂 

Re 3.:
That's fair; thanks for providing additional and helpful context. The `${lib.optionalString …}` interpolation is something I haven't come across yet, so that's probably why I was a bit surprised to see it, proposed a syntax that to me appeared more common in nixpkgs.

Reply:
@afh, is there anything else you'd like to see addressed here?

Reply:
Thanks for the ping, @zaninime, I've decided to direct my time and attention to other matters and kindly request for another person to take over.